### ISSUE TITLE
Missing purchase in successful purchase result

### ISSUE THREAD

**Issue by Lingviston** [android/play-billing-samples#657]
https://github.com/android/play-billing-samples/blob/047d2eb8eef8fe32ec07a115580faea8eb865e43/ClassyTaxiAppKotlin/app/src/main/java/com/example/billing/gpbl/BillingClientLifecycle.kt#L333

Hi
Is the above case "legal"? User has made a purchase, GPBL reports it being successful, but for some reason doesn't provide any "purchase" to process.

In which cases may it happen? How to properly support it?

**Comment by matale** [comment]
Have you actually experienced such a condition? I think this might just be an implementation quirk in the sample otherwise, there is no real reason for the if there other than to log the condition I guess. Not worth worrying about unless you are actually experiencing such an issue in prod.

**Comment by Lingviston** [comment]
Yes, I've experienced it. It seems to happen during subscription cross grade with Deferred replacement mode. Not every time - just must be in certain state. For example they can have subscription, cancelled during trial period.
